lib/bolt/node/winrm.rb

```diff
@@ -337,7 +518,23 @@ def _run_task(task, input_method, arguments)
           # NOTE: cannot redirect STDIN to a .ps1 script inside of PowerShell
           # must create new powershell.exe process like other interpreters
           # fortunately, using PS with stdin input_method should never happen
-          output = execute("try { &""#{remote_path}"" } catch { exit 1 }")
+          output = execute(<<-PS)
```

Comment:
Since `stdin` is nil, that means we're not using the `stdin` input_method. Are the `environment` and `powershell` input methods mutually exclusive or can they be enabled concurrently as will be supported here?

Reply:
I'm going to do some testing around that, especially I want to know what happens if a script is passed the argument hash while it doesn't define the `param ()`, or defines a different set of params than those passed in the hash.

Reply:
Yeah, definitely a useful scenario to think about adding tests for.

For the situation you describe, given extraneous parameters, PowerShell will explode like this when params are splatted (example taken from REPL):

```powershell
PS /Users/Iristyle/source/puppetlabs-dsc_lite> function Foo                                                        >> {                                                                                                               >> [CmdletBinding()]                                                                                               >> param (                                                                                                         >>   [String]                                                                                                      >>   $foo = 'bar'                                                                                                  >> )                                                                                                               >> Write-Host $foo                                                                                                 >> }
PS /Users/Iristyle/source/puppetlabs-dsc_lite> $badParam = @{ 'baz' = 1234 }
PS /Users/Iristyle/source/puppetlabs-dsc_lite> Foo @badParam
Foo : A parameter cannot be found that matches parameter name 'baz'.
At line:1 char:5
+ Foo @badParam
+     ~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Foo], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Foo
```

For *missing* parameters that are not decorated with `[Parameter(Mandatory = $true)]`, standard PowerShell behavior applies - i.e. there are no exceptions thrown for parameter validation issues.

```
PS /Users/Iristyle/source/puppetlabs-dsc_lite> Foo
bar

PS /Users/Iristyle/source/puppetlabs-dsc_lite> Foo -foo hi
hi
```

Reply:
I've made the `environment` & `powershell` methods mutually exclusive, so the parameters hash is only passed to the task when the `powershell` input method is specified.